src/api.v2/model/experiment.js

```diff
@@ -1,22 +1,69 @@
+/* eslint-disable func-names */
 const generateBasicModelFunctions = require('../helpers/generateBasicModelFunctions');
+const sqlClient = require('../../sql/sqlClient');
+const { aggregateIntoJson } = require('../../sql/helpers');
 
 const tableName = 'experiment';
 
-const selectableProps = [
+const experimentFields = [
   'id',
   'name',
   'description',
-  'processing_config',
+  'samples_order',
   'notify_by_email',
   'created_at',
   'updated_at',
 ];
 
 const basicModelFunctions = generateBasicModelFunctions({
   tableName,
-  selectableProps,
+  selectableProps: experimentFields,
 });
 
+const getExperimentData = async (experimentId) => {
+  const sql = sqlClient.get();
+
+  function query() {
+    this.select('*')
+      .from(tableName)
+      .leftJoin('experiment_execution', `${tableName}.id`, 'experiment_execution.experiment_id')
+      .where('id', experimentId)
+      .as('experiment_with_exec');
+  }
+
+  const experimentExecutionFields = [
+    'params_hash', 'state_machine_arn', 'execution_arn',
+  ];
+
+  const result = await aggregateIntoJson(query, experimentFields, experimentExecutionFields, 'pipeline_type', 'pipelines', sql);
+
+  return result[0];
+};
+
+const updateSamplePosition = async (experimentId, oldPosition, newPosition) => {
+  const sql = sqlClient.get();
+
+  // Switches values between the item at newPosition and the one at oldPosition
```

Comment:
is the idea that the initial order of the samples will be added to the database when we call create samples?

Reply:
Right now we don't support batch addition of samples on the api side, we perform one samples create call for each sample we want to upload. The samples are just appended to the end.

Reply:
makes sense